### ISSUE TITLE
High latency and low accuracy with  "huggingface finetuned model converted to faster whisper format"

### ISSUE THREAD

**Issue by bchinnari** [ufal/whisper_streaming#133]
Hi,

I took a Huggingface Hindi language small model (https://huggingface.co/vasista22/whisper-hindi-small) and I fine-tuned it using my data. The fine-tuning is done in batch/offline mode where I have access to full audio file rather than not just chunks (like in a streaming or online fashion). 

I converted the fine-tuned model to faster whisper format. I am getting very good accuracy in offline mode. Now, I am trying to use that model for streaming asr using this repo. 
I observed that latency is high and accuracy is dropping a lot . I also observe there are lot of insertions/hallucinations. 

I have following questions
1. My model is working well in offline mode but not in online mode. I am seeing lot of insertions in "INCOMPLETE" hypothesis part. 
2. I think previous complete hypothesis works as a prompt for the current buffer I guess. Is it possible to remove the prompt when making prediction for the current buffer. Why I ask is that my model is not trained in that fashion. No text is working as prompt in training.

**Comment by Gldkslfmsd** [comment]
> The fine-tuning is done in batch/offline mode where I have access to full audio file rather than not just chunks (like in a streaming or online fashion).

This might be the issue. Try fine-tuning on prefixes.

Moreover, the small model might have small capacity to work on prefixes. Try using it with large chunks, e.g. 5 seconds, it might be better.

Plus make sure you use voice activity controller properly.

Yes, you can deactivate the prompt.

**Comment by zyjcsf** [comment]
> Hi,
> 
> I took a Huggingface Hindi language small model (https://huggingface.co/vasista22/whisper-hindi-small) and I fine-tuned it using my data. The fine-tuning is done in batch/offline mode where I have access to full audio file rather than not just chunks (like in a streaming or online fashion).
> 
> I converted the fine-tuned model to faster whisper format. I am getting very good accuracy in offline mode. Now, I am trying to use that model for streaming asr using this repo. I observed that latency is high and accuracy is dropping a lot . I also observe there are lot of insertions/hallucinations.
> 
> I have following questions
> 
> 1. My model is working well in offline mode but not in online mode. I am seeing lot of insertions in "INCOMPLETE" hypothesis part.
> 2. I think previous complete hypothesis works as a prompt for the current buffer I guess. Is it possible to remove the prompt when making prediction for the current buffer. Why I ask is that my model is not trained in that fashion. No text is working as prompt in training.

Hi， 
May I ask if you have resolved this problem? I'm encountering the same issue. Could you please provide the solution?

**Comment by bchinnari** [comment]
@zyjcsf  ... the issue is not resolved for me.
Why this is happening: The default/pre-trained (faster) whisper models output end time stamp token also for every speech segment in the audio. whisper streaming relies on this end time stamp token.
But the models which are fine-tuned , using huggingface, don't output end time stamp token. Faster whisper did not handle this edge case and produces incorrect (actually more) number of segments and incorrect time stamps. So, whisper streaming processes incorrect (more) segments and producing lot of insertions.
It is discussed at length in https://github.com/SYSTRAN/faster-whisper/issues/1100#issuecomment-2467632243

So, it is the problem with faster whisper rather than whisper streaming.

**Comment by Gldkslfmsd** [comment]
you can try https://github.com/ufal/SimulStreaming , a new generation of whisper_streaming. It uses a better policy AlignAtt. Timestamps are no longer needed for simultaneous mode. Default decoding is without timsestamps.
But there is only one computation backend -- torch as in the original Whisper.
